src/lib.rs

```diff
@@ -507,12 +507,12 @@ impl<'input> fmt::Debug for ExpandedNameOwned<'input> {
 ///
 /// Contains an namespace URI and name pair.
 #[derive(Clone, Copy, PartialEq)]
-pub struct ExpandedName<'input> {
-    uri: Option<&'input str>,
+pub struct ExpandedName<'a, 'input> {
+    uri: Option<&'a str>,
```

Comment:
Why have you changed the lifetime only for `uri`?

Reply:
Because, as far as I understand it, the lifetime of `name` is the lifetime of the initial input, where the lifetime of `uri` corresponds to the lifetime of `ExpandedNameOwned`, since we borrow an owned value from `ExpandedNameOwned` in line 490 `self.ns.as_ref()`. In other words, because `ExpandedNameOwned` owns `ns` (where `uri` is derived from) and only has  a borrow of `name`. Does this make sense?

Reply:
My bad.